### PR TITLE
[FLINK-26300][connectors/common] Increased timeout for LocalstackCont…

### PR DESCRIPTION
**Pull request by CrynetLogistics**
…ainer to start

## What is the purpose of the change

Fix the `LocalstackContainer` startup loop.

Explanation: The localstack container is used to mock aws services so tests can hit a mock endpoint. When we start the container, it occasionally fails with the exception: `IOException - closed before protocol could be determined.` This is what happens whenever we make a request (e.g. read from mock s3) prior to the container being ready**. A similar issue was resolved in Kinesalite by increasing the timeout from 1s to 10s, but Localstack is a larger box. So here we increase the wait to 30s.

** This is a known limitation, please see issue [#1202](https://github.com/localstack/localstack/issues/1202) in Localstack/localstack or this [blog post](https://ignas.me/tech/waiting-localstack-s3-start/).

## Brief change log

Increased timeout of container startup to 30s from 10s.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / no)n
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / no)n
  - The serializers: (yes / no / don't know)n
  - The runtime per-record code paths (performance sensitive): (yes / no / don't know)n
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / no / don't know)n
  - The S3 file system connector: (yes / no / don't know)n

## Documentation

  - Does this pull request introduce a new feature? (yes / no)n
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)n

### CODE REVIEW

**Comment by CrynetLogistics** [comment]
@flinkbot azure run

**Comment by CrynetLogistics** [comment]
@flinkbot run azure

File: flink-connectors/flink-connector-aws-base/src/test/java/org/apache/flink/connector/aws/testutils/LocalstackContainer.java [hunk 1]
```diff
@@ -62,7 +62,7 @@ public String getEndpoint() {
         @Override
         protected void waitUntilReady() {
             try {
-                Thread.sleep(10000);
+                Thread.sleep(30_000);
```
Comment (dannycranmer) [inline flink-connectors/flink-connector-aws-base/src/test/java/org/apache/flink/connector/aws/testutils/LocalstackContainer.java:65]: Please justify the reason for this.

Is there any other way we can establish if the test stack is ready?

Comment (CrynetLogistics) [reply flink-connectors/flink-connector-aws-base/src/test/java/org/apache/flink/connector/aws/testutils/LocalstackContainer.java:65]: I couldn't see a good way to poll whether the container is ready. On the `testcontainers` side, I tried tracking the output of `isRunning()`/`isCreated()`/`isReusable()` through successful and unsuccessful runs and none of them are telling. Anything on the `localstack` api side puts it in a funny state in the same way the s3bucket read strategy does.

**Comment by CrynetLogistics** [comment]
@flinkbot run azure

**Comment by CrynetLogistics** [comment]
@flinkbot run azure

**Comment by CrynetLogistics** [comment]
@flinkbot run azure

File: flink-connectors/flink-connector-aws-base/src/test/java/org/apache/flink/connector/aws/testutils/LocalstackContainer.java (hunk 1 above)
Comment (dannycranmer) [reply flink-connectors/flink-connector-aws-base/src/test/java/org/apache/flink/connector/aws/testutils/LocalstackContainer.java:65]: There is a solution here: https://ignas.me/tech/waiting-localstack-s3-start/

> I’ve noticed that hitting Refresh button in LocalStack Dashboard calls /graph endpoint to know if and how many nodes (mock services) are available, i.e. healthy and ready to accept incoming requests. Bullseye!

**Review by dannycranmer — APPROVED** [review]

File: flink-connectors/flink-connector-aws-base/src/test/java/org/apache/flink/connector/aws/testutils/LocalstackContainer.java (hunk 1 above)
Comment (CrynetLogistics) [reply flink-connectors/flink-connector-aws-base/src/test/java/org/apache/flink/connector/aws/testutils/LocalstackContainer.java:65]: Thanks, I have created this: https://issues.apache.org/jira/browse/FLINK-26309.

**Pull request merged**